src/executor/infer_graph_attr_pass.cc

```diff
@@ -302,18 +341,27 @@ inline bool SameType(const nnvm::NodeAttrs& attrs,
   return true;
 }
 
-// assigning default type N to both input and output attrs with value -1
-template <int default_val, int none>
-inline bool DefaultType(const nnvm::NodeAttrs& attrs,
-                        const Context& ctx,
-                        std::vector<int> *iattr,
-                        std::vector<int> *oattr) {
-  // TODO(junwu): check whether need to use ctx
+inline bool DefaultStorageType(const nnvm::NodeAttrs& attrs,
+                               const int dev_mask,
+                               int* dispatch_mode,
+                               std::vector<int> *iattr,
+                               std::vector<int> *oattr) {
+  bool fallback = false;
   for (int& v : *oattr) {
-    if (v == none) v = default_val;
+    if (v == -1) v = kDefaultStorage;
+    if (v != kDefaultStorage) fallback = true;
   }
   for (int& v : *iattr) {
```

Comment:
int& -> int

Reply:
As discussed offline, not really necessary to change it in a for loop.